build/references/coreth-arc20s.md

```diff
@@ -0,0 +1,103 @@
+
+##### assetCall
+
+`assetCall` is a precompiled contract at address `0xassetCallAddr`. `assetCall` allows users to atomically transfer a native asset to a given address and (optionally) also make a contract call to that same address. This is parallel to how a normal transaction can send value to some address, and atomically call that address with some `data`.
+
+
+```text
+assetCall(address addr, uint256 assetID, uint256 assetAmount, bytes memory callData) -> {ret: bytes memory}
+```
+
+These arguments can be packed by `abi.encodePacked(...)` in Solidity since there is only one argument with variadic length (callData). The first three arguments are constant length, so the precompiled contract simply parses the call `data` input as:
+
+
+```text
++-------------+---------------+--------------------------------+
+| address     : address       |                       20 bytes |
++-------------+---------------+--------------------------------+
+| assetID     : uint256       |                       32 bytes |
++-------------+---------------+--------------------------------+
+| assetAmount : uint256       |                       32 bytes |
++-------------+---------------+--------------------------------+
+| callData    : bytes memory  |            len(callData) bytes |
++-------------+---------------+--------------------------------+
+                              |       84 + len(callData) bytes |
+                              +--------------------------------+
+```
+
+##### assetBalance
+
+`assetBalance` is a precompiled contract at address `0xassetBalance`. `assetBalance` is the ANT equivalent of using `balance` to get the AVAX balance.
+
+```text
+assetBalance(address addr, uint256 assetID) -> {balance: uint256}
+```
+
+These arguments can be packed by `abi.encodePacked(...)` in Solidity since all of the arguments have constant length.
+
+```text
++-------------+---------------+-----------------+
+| address     : address       |        20 bytes |
++-------------+---------------+-----------------+
+| assetID     : uint256       |        32 bytes |
++-------------+---------------+-----------------+
+                              |        52 bytes |
+                              +-----------------+
+```
+
+## ARC20s on the C-Chain
+
+An ARC20 is the Avalanche equivalent of an ERC20. An ERC20 is just a smart contract that maintains balances for accounts and provides an incredibly useful, standardized interface for smart contract developers.
+
+ERC20s can be deployed directly to the C-Chain without modification, but these contracts do not have a native asset equivalent. Therefore, we create `Wrapped ARC20s` (`WARC20s`) which wrap native assets into an ERC20 interface, so that `ANTs` can be used following the same interface as an ERC20.
```

Comment:
I think you got everything right here, but I would love a Confluence page that outlines our style guide for ERC20 vs ARC20 usage. Also, technically it is ERC-20.

Reply:
Ok, I'll change ERC20 -> ERC-20 and ARC20 -> ARC-20